From 9acb771798c8efc2177425bd9b1ad6b52a1dce6a Mon Sep 17 00:00:00 2001
From: Radical <radical@radical.fun>
Date: Wed, 5 Aug 2026 18:21:18 +0200
Subject: [PATCH] add drm lease functions

---
 .../gamescope-session-plus/gamescope-session-plus      | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/usr/share/gamescope-session-plus/gamescope-session-plus b/usr/share/gamescope-session-plus/gamescope-session-plus
index 8a89c01..d2de9d4 100755
--- a/usr/share/gamescope-session-plus/gamescope-session-plus
+++ b/usr/share/gamescope-session-plus/gamescope-session-plus
@@ -237,6 +237,16 @@ if [ -z "$GAMESCOPECMD" ]; then
 		FAKE_OUTPUT_MM_OPTION="--fake-output-mm $FAKE_OUTPUT_MM"
 	fi
 
+	LEASE_CONNECTOR_OPTION=""
+	if [ -n "$ARMADA_LEASE_CONNECTOR" ] && gamescope_has_option "--lease-connector"; then
+		LEASE_CONNECTOR_OPTION="--lease-connector $ARMADA_LEASE_CONNECTOR"
+	fi
+
+	IGNORE_TOUCH_DEVICE_OPTION=""
+	if [ -n "$ARMADA_IGNORE_TOUCH_DEVICE" ] && gamescope_has_option "--ignore-touch-device"; then
+		IGNORE_TOUCH_DEVICE_OPTION="--ignore-touch-device $ARMADA_IGNORE_TOUCH_DEVICE"
+	fi
+
 	HDR_OPTIONS=""
 	if [ "$ENABLE_GAMESCOPE_HDR" == "1" ] && [ "$ENABLE_GAMESCOPE_WSI" == "1" ] && gamescope_has_option "--hdr-enabled" && gamescope_has_option "--hdr-itm-enable"; then
 		HDR_OPTIONS="--hdr-enabled --hdr-itm-enable"
-- 
2.55.0

